pkg/controllers/user/helm/controller.go

```diff
@@ -346,15 +354,30 @@ func (l *Lifecycle) createAppRevision(obj *v3.App, template, notes string, faile
 
 func (l *Lifecycle) writeKubeConfig(obj *v3.App, tempDir string, remove bool) (string, error) {
 	var token string
+	var err error
+	var user *mgmtv3.User
 
-	userID := obj.Annotations["field.cattle.io/creatorId"]
-	user, err := l.UserClient.Get(userID, metav1.GetOptions{})
-	if err != nil && !errors.IsNotFound(err) {
+	isSystemProject, err := l.isSystemProject(obj)
+	if err != nil {
 		return "", err
-	} else if errors.IsNotFound(err) && remove {
-		token, err = l.SystemAccountManager.GetOrCreateProjectSystemToken(obj.Namespace)
-	} else if err == nil {
-		token, err = l.UserManager.EnsureToken(helmTokenPrefix+user.Name, description, user.Name)
+	}
+
+	if l.isSystemComponent(obj) {
+		if isSystemProject { // If the app is being deployed into system project, we will create a cluster admin token for it.
+			token, err = l.SystemAccountManager.GetOrCreateClusterSystemToken(l.ClusterName)
+		} else { // If the app is being deployed into user project, we will create a project member token for it.
```

Comment:
`isSystemComponent` will always check if the app being installed is `cluster-monitoring` and in the namespace `cattle-prometheus`. This ns will always be in system project, I'm not sure I get why we need to check `isSystemProject` again?